Content.Server/GameObjects/Components/Items/Storage/ServerStorageComponent.cs

```diff
@@ -469,6 +492,95 @@ void IActivate.Activate(ActivateEventArgs eventArgs)
+        /// <returns></returns>
+        async Task IAfterInteract.AfterInteract(AfterInteractEventArgs eventArgs)
+        {
+            if (!_quickInsert) return;
+            if (!eventArgs.InRangeUnobstructed(ignoreInsideBlocker: true, popup: true)) return;
+
+            // Pick up all entities in a radius around the clicked location.
+            // The last half of the if is because carpets exist and this is terrible
+            if(eventArgs.Target == null || !eventArgs.Target.HasComponent<StorableComponent>())
+            {
+                var validStorables = new List<IEntity>();
+                foreach (var entity in Owner.EntityManager.GetEntitiesInRange(eventArgs.ClickLocation, 1))
+                {
+                    if (!entity.Transform.IsMapTransform
+                        || entity == eventArgs.User
+                        || !entity.HasComponent<StorableComponent>())
+                        continue;
+                    validStorables.Add(entity);
+                }
+
+                //If there's only one then let's be generous
+                if (validStorables.Count > 1)
+                {
+                    var doAfterSystem = EntitySystem.Get<DoAfterSystem>();
+                    var doAfterArgs = new DoAfterEventArgs(eventArgs.User, 0.2f * validStorables.Count, CancellationToken.None, Owner)
+                    {
+                        BreakOnStun = true,
+                        BreakOnDamage = true,
+                        BreakOnUserMove = true,
+                        NeedHand = true,
+                    };
+                    var result = await doAfterSystem.DoAfter(doAfterArgs);
+                    if (result != DoAfterStatus.Finished) return;
+                }
+
+                var successfullyInserted = new List<IEntity>();
+                var successfullyInsertedPositions = new List<EntityCoordinates>();
+                foreach (var entity in validStorables)
+                {
+                    // Check again, situation may have changed for some entities, but we'll still pick up any that are valid
+                    if (!entity.Transform.IsMapTransform
+                        || entity == eventArgs.User
+                        || !entity.HasComponent<StorableComponent>())
+                        continue;
+                    var coords = entity.Transform.Coordinates;
+                    if (PlayerInsertEntityInWorld(eventArgs.User, entity))
+                    {
+                        successfullyInserted.Add(entity);
+                        successfullyInsertedPositions.Add(coords);
+                    }
+                }
+
+                // If we picked up atleast one thing, do a cool animation!
+                if (successfullyInserted.Count>0)
+                {
+                    var entityIds = new EntityUid[successfullyInserted.Count];
+                    var positions = new EntityCoordinates[successfullyInserted.Count];
```

Comment:
why copy the entire positionsarray?

Reply:
Multiple item positions, they all animate from their respective positions.

As to why I didn't use `ToArray()` instead... I don't actually know 😅

Reply:
im aware you need all those positions, im just confused why you are allocating another array

Reply:
I have no previous array, I have a list.

I guess those serialise fine though so, guess I was being paranoid.